libraries/tools/kotlin-gradle-plugin/src/functionalTest/kotlin/org/jetbrains/kotlin/gradle/unitTests/KotlinTargetHierarchyDslTest.kt

```diff
@@ -319,6 +320,69 @@ class KotlinTargetHierarchyDslTest {
         )
     }
 
+    @Test
+    fun `test - hierarchy js and wasm`() {
+        kotlin.targetHierarchy.custom {
+            common {
+                group("web") {
+                    withJs()
+                    withWasm()
+                }
+            }
+        }
+
+        @OptIn(ExperimentalWasmDsl::class)
+        kotlin.wasm()
+        kotlin.js()
+
+        assertEquals(
+            stringSetOf("webMain", "jsMain", "wasmMain"),
+            kotlin.dependingSourceSetNames("commonMain")
+        )
+
+        assertEquals(
+            stringSetOf("jsMain", "wasmMain"),
+            kotlin.dependingSourceSetNames("webMain")
+        )
+    }
+
+    @Test
+    fun `test - hierarchy js and wasm split`() {
+        kotlin.targetHierarchy.custom {
+            common {
+                group("jsAndJvm") {
+                    withJs()
+                    withJvm()
+                }
+                group("wasmAndLinux") {
+                    withWasm()
+                    withLinuxX64()
+                }
+            }
+        }
+
+        @OptIn(ExperimentalWasmDsl::class)
```

Comment:
NIT: Feel free to just lift this to a file level 👍

Reply:
done